### PR TITLE
eng, ignore "generated" folder in sdk sync, if that folder not in sdk repo

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**

### CODE REVIEW

File: eng/sdk/sync_sdk.py [hunk 1]
```diff
@@ -44,26 +47,45 @@ def update_emitter(version: str):
...
             os.path.join(sdk_root, 'eng/common/scripts/TypeSpec-Project-Sync.ps1'),
-            artifact_path
+            tsp_location_path
         ]
         subprocess.check_call(cmd, cwd=sdk_root)
 
         cmd[1] = os.path.join(sdk_root, 'eng/common/scripts/TypeSpec-Project-Generate.ps1')
         subprocess.check_call(cmd, cwd=sdk_root)
 
+        if not generated_samples_exists:
```
Comment (haolingdong-msft) [inline eng/sdk/sync_sdk.py:84]: What is the case where `generated folder` not in sdk repo?

Comment (weidongxu-microsoft) [reply eng/sdk/sync_sdk.py:84]: service can generate it, copy some, then delete the "generated" folder.

in some case, some string in there would trigger cspell.

Comment (weidongxu-microsoft) [reply eng/sdk/sync_sdk.py:84]: But when service generate next api-version, regen these test/sample still helpful.

Comment (haolingdong-msft) [reply eng/sdk/sync_sdk.py:84]: I see. Thanks

**Pull request merged**